braid/drive.c

```diff
@@ -648,13 +648,39 @@ _braid_Drive(braid_Core  core,
       _braid_FRestrict(core, level);
    }
 
-   /* Allow final access to Braid by carrying out an F-relax to generate points */
-   /* Record it only if sequential time stepping */
-   if (max_levels > 1)
+
+   /* Allow final access to Braid by carrying out a relaxation to generate points */
+   if ( _braid_CoreElt(core, finalFCrelax) )
```

Comment:
@rfalgout @steffi7574 @eric-c-cyr 
Here is the reworked logic about when to call the final FCRelax or the final FAccess.  Right now, only FCRelax OR FAccess is called at the end of a Braid run.  This saves work.

But this does mean that FCRelax needs to do everything FAccess does, and so I've added User-Access hooks into FCRelax, including at time t==0, which needed some extra logic.

A couple more notes, if max_levels is 1, we now rely on FCRelax to do the serial solution in time.  Eric and Stefanie may want to double check this works with everything on their end.  For instance, both Access and FCRelax should now store ulast if done==1.  The unit tests in TorchBraid pass, at least.  But double checking that everything works for max_levels==1 would be good.